std/array.d

```diff
@@ -708,8 +708,12 @@ if (isDynamicArray!T && allSatisfy!(isIntegral, I))
 
     auto arr = minimallyInitializedArray!(int[])(42);
     assert(arr.length == 42);
-    // Elements aren't necessarily initialized to 0
-    assert(!arr.equal(0.repeat(42)));
+
+    // Elements aren't necessarily initialized to 0, so don't do this:
+    // assert(arr.equal(0.repeat(42)));
+    // If that is needed, initialize array like this instead:
+    auto arr2 = 0.repeat(42).array;
```

Comment:
Is there a reason to recommend this instead of `new int[42]`?

Reply:
No, but I didn't consider it worse either, unless I count the fact that `new int[42]` is closer to `minimallyInitializedArray!int(42)` than my line is.

Reply:
In that case I think lines 712 through 715 should be replace with simply:
```d
    // Because an int does not contain any pointers the elements of `arr`
    // are not initialized to zero.
```
But if you disagree this can still go forward.

Reply:
Mathias wanted an example of how to initialize the array correctly, so I'm not deleting it.

I changed the initialization to what you suggested instead.